src/collection-browser.ts

```diff
@@ -1211,10 +1211,19 @@ export class CollectionBrowser
       this.preloadCollectionNames(results);
       this.updateDataSource(pageNumber, results);
     }
-    if (results.length < this.pageSize) {
+
+    // When we reach the end of the data, we can't always immediately cap off the
+    // tile count, because there may be previous pages that haven't finished loading
+    // yet. So we flag the end of the data, but let the final page load set the count.
+    // This avoids race conditions dependent on the order pages arrive in.
+    if (results.length < this.pageSize || this.endOfDataReached) {
       this.endOfDataReached = true;
-      // this updates the infinite scroller to show the actual size
-      if (this.infiniteScroller) {
+      // This updates the infinite scroller to show the actual size
+      // (provided there are no other pages still being fetched)
+      if (
+        this.infiniteScroller &&
+        Object.keys(this.pageFetchesInProgress).length === 1 // This must be the last page fetch
```

Comment:
Q: is `Object.keys(this.pageFetchesInProgress).length === 1` equal to `results.length < this.pageSize || this.endOfDataReached`?

Reply:
The `Object.keys(this.pageFetchesInProgress).length === 1` condition ensures that no other page fetches are pending, because we don't want to prematurely set the infinite scroller's final item count before all remaining fetches have finished.

The earlier `results.length < this.pageSize || this.endOfDataReached` condition is checking whether we've reached the last page of results. `results.length < this.pageSize` will be true if the page we're currently processing is the last one (or beyond), while `this.endOfDataReached` will be true if we've _previously_ processed the last page (e.g., if the page responses have arrived out of order).

Reply:
Although now I'm wondering why the infinite scroller's item count was set in such a roundabout way, despite having the total result count present on every response (even going back to the `advancedsearch.php` days...)